Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1835 @@
+  id o = nil;
+
+  while ((o = [en nextObject]) != nil)
+    {
+      [result addObject: o];
+    }
+
+  return GS_IMMUTABLE(result);
+}
+@end
+
+// Mutable Ordered Set
+@implementation NSMutableOrderedSet
+// Creating a Mutable Ordered Set
++ (void) initialize
+{
+  if (self == [NSMutableOrderedSet class])
+    {
+      NSMutableOrderedSet_abstract_class = self;
+      NSMutableOrderedSet_concrete_class = [GSMutableOrderedSet class];
+    }
+}
+
++ (id) allocWithZone: (NSZone*)z
+{
+  if (self == NSMutableOrderedSet_abstract_class)
+    {
+      return NSAllocateObject(NSMutableOrderedSet_concrete_class, 0, z);
+    }
+  else
+    {
+      return NSAllocateObject(self, 0, z);
+    }
+}
+
++ (instancetype) orderedSetWithCapacity: (NSUInteger)capacity
+{
+  return AUTORELEASE([[self allocWithZone: NSDefaultMallocZone()] initWithCapacity: capacity]);
+}
+
+- (Class) classForCoder
+{
+  return NSMutableOrderedSet_abstract_class;
+}
+
+- (instancetype) initWithCapacity: (NSUInteger)capacity
+{
+  self = [self init];
+  return self;
+}
+
+- (instancetype) init
+{
+  self = [super init];
+  if(self == nil)
+    {
+      NSLog(@"Could not init class");
+    }
+  return self;
+}
+
+- (void) addObject: (id)anObject
+{
+  [self insertObject: anObject atIndex: [self count]];
+}
+
+- (void) addObjects: (const id[])objects count: (NSUInteger)count
+{
+  NSUInteger i = 0;
+  for (i = 0; i < count; i++)
+    {
+      id obj = objects[i];
+      [self addObject: obj];
+    }
+}
+
+- (void) addObjectsFromArray: (NSArray *)otherArray
+{
+  NSEnumerator *en = [otherArray objectEnumerator];
+  id obj = nil;
+  while((obj = [en nextObject]) != nil)
+    {
+      [self addObject: obj];
+    }
+}
+
+- (void) insertObject: (id)object atIndex: (NSUInteger)index  // required override
+{
+  [self subclassResponsibility: _cmd];
+}
+
+- (void) setObject: (id)object atIndexedSubscript: (NSUInteger)index
+{
+  if ([self count] == index)
+    {
+      [self addObject: object];
+    }
+  else
+    {
+      [self replaceObjectAtIndex: index withObject: object];
+    }
+}
+
+- (void) insertObjects: (NSArray *)array atIndexes: (NSIndexSet *)indexes
+{
+  NSUInteger	index = [indexes firstIndex];
+  NSEnumerator	*enumerator = [array objectEnumerator];
+  id		object = [enumerator nextObject];
+
+  while (object != nil && index != NSNotFound)
+    {
+      [self insertObject: object atIndex: index];
+      object = [enumerator nextObject];
+      index = [indexes indexGreaterThanIndex: index];
+    }
+}
+
+- (void) removeObject: (id)anObject
+{
+  NSUInteger index;
+
+  if (anObject == nil)
+    {
+      NSWarnMLog(@"attempt to remove nil object");
+      return;
+    }
+
+  index = [self indexOfObject: anObject];
+  if (NSNotFound != index)
+    {
+      [self removeObjectAtIndex: index];
+    }
+}
+
+- (void) removeObjectAtIndex: (NSUInteger)index  // required override
+{
+  [self subclassResponsibility: _cmd];
+}
+
+- (void) _removeObjectsFromIndices: (NSUInteger*)indices
```

Comment:
Why did you reintroduce this method? It is no longer used anywhere in this class and if there isn't a demonstrable case that it gets used somewhere else in public code there is no need for it.

Reply:
After my comment you introduced the usage of this method. This really is frustrating. I wanted to help you get this polished and out to the world. I rather wait until you are through with your days work so we don't overlap.